File: bg/okgame/legacy-src/src/Engine/state/StateManager.cpp
#include "stdafx.h"

//------------------------------------------------------------------------------
//Copyright Robert Pelloni.
//All Rights Reserved.
//------------------------------------------------------------------------------


//#pragma once




Logger BobStateManager::log = Logger("BobStateManager");


BobStateManager::BobStateManager()
{ //=========================================================================================================================
}

shared_ptr<Engine> BobStateManager::getCurrentState()
{ //=========================================================================================================================

	if(states.size()>0)
	return states.get(0);
	else return nullptr;
}

//void BobStateManager::setState(Engine* s)
//{ //=========================================================================================================================
//	currentState = s;
//}

void BobStateManager::pushState(shared_ptr<Engine> s)
{ //=========================================================================================================================
	states.insert(0, s);
}

void BobStateManager::popState()
{ //=========================================================================================================================
	if(states.size()>0)
	states.removeAt(0);
}

void BobStateManager::update()
{
	Engine::updateTimers();
	getCurrentState()->update();
}

void BobStateManager::render()
{
	getCurrentState()->render();
}

File: bg/bobsgameonlinejava/src/main/java/com/bobsgame/client/state/StateManager.java
package com.bobsgame.client.state;

public class StateManager {
	public State currentState;

	public StateManager() {

	}

	public State getState() {
		return currentState;
	}

	public void setState(State s) {
		currentState = s;
	}

	public void update() {
		currentState.update();
	}

	public void render() {
		currentState.render();
	}
}